ironfist/src/cpp/game/save.cpp

```diff
@@ -406,18 +447,34 @@ void game::LoadGame(char* filnam, int newGame, int a3) {
             gpGame->ResetIronfistGameState();
 
 			if (mp->script().present()) {
-				ScriptingInitFromString(mp->script().get().c_str());
+				ScriptingInitFromString(mp->script().get());
 			}
 
-			i = 0;
+			std::map<std::string, mapVariable> mapVariables;
 			for (ironfist_map::map_t::mapVariable_const_iterator it = mp->mapVariable().begin();
-			it != mp->mapVariable().end();
-				it++, i++) {
-				SetMapVariables(it->id().get().c_str(), it->value().get().c_str());
+				it != mp->mapVariable().end(); it++) {
+				mapVariable *mapVar = new mapVariable;
+				std::string mapVariableId = it->id().get();
+				MapVarType mapVariableType = StringToMapVarType(it->type());
+				mapVar->type = mapVariableType;
+				if (isTable(mapVariableType)) {
+					luaTable *lt = new luaTable;
+					ReadMapVarXML(*lt, it->table().get());
+					mapVar->tableValue = lt;
+				} else if (isStringNumBool(mapVariableType)) {
+					std::string *sV = new std::string;
+					*sV = it->value().get();
+					mapVar->singleValue = sV;
+				} else {
+					ErrorLoadingMapVariable(mapVariableId, " A map variable can only be a table, number, string or boolean.");
+				}
+				mapVariables[mapVariableId] = *mapVar;
 			}
+			WriteMapVariablesToLUA(mapVariables);
+

```

Comment:
To clarify: Is it still crashing after these changes?

Because you are allocating the variables with new instead of on the stack, the destructor should not be called until they are explicitly deleted.